### ISSUE TITLE
large int values get converted to floats when parsing json and outputting yaml

### ISSUE THREAD

**Issue by jakkanen** [mikefarah/yq#1756]
**Describe the bug**

int values with 7 or more digits become floats when parsed as json and outputted as yaml

Version of yq: 4.35.1
Operating system: mac
Installed via: homebrew

**Input Yaml**

data1.yml:
```yaml
[999999, 1000000, 1000001]
```

**Command**
The command you ran:
```
yq -p json -o yaml data1.yaml 
```

**Actual behavior**

```yaml
- 999999
- 1e+06
- 1.000001e+06
```

**Expected behavior**

```yaml
- 999999
- 1000000
- 1000001
```

**Additional context**
not seeing ints getting mangled with `-p yaml -o yaml`, `-p yaml -o json`, or `-p json -o json`

**Comment by mikefarah** [comment]
Thanks for raising - problem was that the JSON decoder would parse those numbers as floats. Will be fixed in the next release.

**Comment by mikefarah** [comment]
Fixed in 4.35.2
